.github/workflows/get_rocks_modified_and_build_scan_test.yaml

```diff
@@ -0,0 +1,51 @@
+name: Get ROCKs modified and build-scan-test them
+
+on:
+  workflow_call:
+
+jobs:
+  get-all-rock-paths:
+    name: Get all rockcraft.yaml paths
+    runs-on: ubuntu-22.04
+    outputs:
+      paths: ${{ steps.find.outputs.paths}}
+    steps:
+      - name: Checkout repository code
+        uses: actions/checkout@v3
+      - name: Get all rockcraft.yaml paths using find
+        id: find
+        run: |
+          paths=$(find ./ -name "rockcraft.yaml" -printf "%h:\n  - '%h/**'\n" | sed 's/\.\///g')
+          echo "Paths found:"
+          echo "${paths}"
+          # Passing multi-line string to output requires the use of a delimiter
+          echo "paths<<EOF" >> $GITHUB_OUTPUT
+          echo "${paths}" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
+  changes:
+    name: Get paths for ROCKs modified by PR
+    needs: get-all-rock-paths
+    runs-on: ubuntu-22.04
+    permissions:
+      pull-requests: read
+    # Set job outputs to values from filter step
+    outputs:
+      rock_paths: ${{ steps.filter.outputs.changes }}
+    steps:
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: ${{ needs.get-all-rock-paths.outputs.paths}}
+
+  build-scan-test-rock:
+    name: Build, scan and test
+    needs: changes
+    strategy:
+      fail-fast: false
+      matrix:
+        rock-dir: ${{ fromJson(needs.changes.outputs.rock_paths) }}
+    uses: ./.github/workflows/build_scan_and_test_rock.yaml
+    secrets: inherit
+    with:
+      rock-dir: ${{ matrix.rock-dir }}
```

Comment:
Should we split this part out into a separate workflow?  I'm thinking:

* get_rocks_modified.yaml
  * outputs the rocks modified in a json string that can be consumed
* get_rocks_modified_and_build_scan_test.yaml
  * uses the previous yaml

The get_rocks_modified part at least feels reusable for other cases, although I can't think of what right away

Reply:
So I pushed a commit. I also pushed the same changes [here](https://github.com/canonical/seldonio-rocks/pull/65) in order to test them but CI is waiting

Reply:
This could be reusable for on_push events (although I 'm not sure if it will work out of the box with `on_push` events or we will need to extend it)

Reply:
I think the CI looks like its working, but the test (or really, build) itself failed?

Reply:
RE build, those `mlserver` need to be fixed at the moment, they fail locally too. Regarding the other one, I need to take a look if we set Trivy action up the right way.